firebase-installations/src/androidTest/java/com/google/firebase/installations/FirebaseInstallationsInstrumentedTest.java

```diff
@@ -266,6 +268,73 @@ public void testGetId_PersistedFidOk_BackendError() throws Exception {
         .isEqualTo(PersistedFid.RegistrationStatus.REGISTER_ERROR);
   }
 
+  @Test
+  public void testGetId_fidRegistrationUncheckedException_statusUpdated() throws Exception {
+    // Mocking unchecked exception on FIS createFirebaseInstallation
+    when(mockClient.createFirebaseInstallation(anyString(), anyString(), anyString(), anyString()))
+        .thenAnswer(
```

Comment:
will doThrow not work in this case?

Reply:
thenThrow and doThrow only stubs checked exceptions.
https://github.com/mockito/mockito/issues/1155
https://github.com/mockito/mockito/issues/1293#issuecomment-390921430